### PR TITLE
Carbon Dismemberment , second attempt.

### LINKED ISSUE #18532: Beheading husks regrows hair, presumably creates implantable head
Cross-posting from https://github.com/HippieStationCode/HippieStation13/issues/2516. The head is still obviously that of a husk (except for the magically regenerating hair); I assume it can also be transplanted but have not gotten a chance to test it

### LINKED ISSUE #18584: Ling Stasis appearance bug
Changeling regen stasis doesn't fix the "burnt husk" look that you get after getting too much burn damage
Transforming doesn't fix it either
Terrible for stealth as everyone will know you're either badmin shennanigan spawn or a changeling or some other antag.

### PR DESCRIPTION
**Pull request by phil235**
:cl: phil235
add: Monkeys can be dismembered.
add: Humans, monkeys and aliens can drop limbs when they are gibbed.
add: Roboticists can remove flashes, wires and power cells that they inserted in robot head or torso by using a crowbar on it.
add: visual wounds appear on a monkey's bodyparts when injured, exactly like humans.
add: Using a health analyzer on a monkey or alien shows the injuries to each bodypart.
add: Monkeys can become husks.
fix: Human-monkey transformations now respect missing limbs. No more limb regrowth by becoming a monkey.
fix: Changeling's regenerate ability also work while in monkey form.
fix: Alien larva has its own gib and dust animation.
fix: A bodypart that looks wounded still looks bloody when dismembered.
fix: Amputation surgery now works on robotic limbs, and monkeys.
/:cl:
- I rearranged X_defense.dm mob files, more damage_procs.dm. Here's what's inside:
  - X_defense.dm: is for the procs of attacks onto the mob, all the XXX_act() proc (things happening to the mob), as well as protection check and get procs (armor, ear prot, projectile dismemberment)
  - damage_procs.dm: actual damage procs like adjustBruteLoss() getfireloss, any proc that handles damaging.
- some bugfixes with gibspawner effects.
- monkey's bodyparts can be dismembered and are used to create its icon.
- <b>brains are no longer carbons.</b> :tada: 
- all carbon have bodyparts that can be dropped when the mob is gibbed.
- adminspawned bodyparts now have a default icon.
- robotic parts are now a child of bodyparts.
- health analyzer on alien/monkey shows damage on each limb
- added admin option to add/remove/augment bodyparts for all carbon (instead of just remove on humans)
- Fixes keycheck message spam for janicart and all when trying to move.
- Fixes bug with buckling to a scooter while limbless.
- removed arg "hit_zone" in proj's on_hit() because we can already use the def_zone var (where hit_zone got its value)
- Fixes mob not getting any damage when hit by a projectile on their missing limb, despite a hit message shown). carbon/apply_damage() now when we specify a def_zone and the corresponding BP is missing we default to the chest instead of stopping the proc. Consistently with how human/attacked_by() default to its attack to chest if missing limb.
- Fixes mini uzi icon when empty and no mag (typo).
- I renamed and changed a bit check_eye_prot and ear prot
- renamed flash_eyes to flash_act()
- I made a soundbang_act() similar to flash_act but for loud bangs.
- added a gib and dust animation to larva.
- husked monkeys
- no damage overlay for husk or skeleton.
- damage overlay for robotic limb now.
- no damage overlay when organic bodypart husked.
- one handed human with a bloody hand still get a bloody single hand overlay.
- fix admin heal being unable to heal robotic bodyparts.
- slightly touched robotic bodypart sprites (head one pixel too high)
- Fixes #18532 "beheaded husk has hair".
- Fixes #18584 "Ling stasis appearance bug"
- no more eyes or lipstick on husks.
- can remove flashes/wires/cells from robot chest and head with crowbar.
- Fixes not being able to surgically amputate robotic arm/leg.
- dropped bodyparts keep the damage overlays they had while attached.
- human-monkey transformation now respects missing limbs. No more limb regrowth by becoming a monkey.
- Changeling's regenerate ability also work while in monkey form.
- NODISMEMBER species can still lack a limb if they were another species with a missing limb and got species-changed.
- Admin heal regenerates limbs on all carbons.

### CODE REVIEW

**Comment by Shadowlight213** [comment]
3many bulletpoints. ahhhhhh

File: code/_onclick/item_attack.dm [hunk 1]
```diff
@@ -90,7 +100,7 @@
 	if(I.attack_verb && I.attack_verb.len)
 		message_verb = "[pick(I.attack_verb)]"
 	else if(!I.force)
-		return 0
+		return
```
Comment (optimumtact) [inline code/_onclick/item_attack.dm]: null instead of what I assume is a false ?

File: code/modules/mob/living/carbon/damage_procs.dm [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	var/list/obj/item/bodypart/parts = get_damageable_bodyparts()
+	if(!parts.len)
+		return
+	var/obj/item/bodypart/picked = pick(parts)
+	if(picked.take_damage(brute,burn))
+		update_damage_overlays()
+
+
+//Heal MANY bodyparts, in random order
+/mob/living/carbon/heal_overall_damage(brute, burn, only_robotic = 0, only_organic = 1, updating_health = 1)
```
Comment (Ergovisavi) [inline code/modules/mob/living/carbon/damage_procs.dm]: On the testmerged sybil server, this broke. Lightgeists use heal_overall_damage to heal people, and any humans are completely unable to be healed by them.

It's also apparently one of the very few uses of heal_overall_damage, but hey!

File: code/_onclick/item_attack.dm (hunk 1 above)
Comment (phil235) [inline code/_onclick/item_attack.dm]: The return value of that proc is no longer used for anything. I think leaving a "return 0" could confuse readers thinking the "0" value has some significance.

File: code/modules/mob/living/carbon/damage_procs.dm (hunk 2 above)
Comment (phil235) [inline code/modules/mob/living/carbon/damage_procs.dm]: Correct, it's a case of the argument of the proc being changed and the lightgeist code using the old args. I'll fix.

File: code/modules/mob/living/carbon/damage_procs.dm [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+/mob/living/carbon/take_overall_damage(brute, burn, updating_health = 1)
+	if(status_flags & GODMODE)
+		return	//godmode
+
+	var/list/obj/item/bodypart/parts = get_damageable_bodyparts()
+	var/update = 0
+	while(parts.len && (brute>0 || burn>0) )
+		var/obj/item/bodypart/picked = pick(parts)
+		var/brute_per_part = round(brute/parts.len, 0.01)
+		var/burn_per_part = round(burn/parts.len, 0.01)
```
Comment (phil235) [inline code/modules/mob/living/carbon/damage_procs.dm]: one thing I forgot to mention: we round the dmg on each part to 0.01 so we don't get huge decimal numbers when we split some damage between each bodypart.

File: code/_onclick/hud/human.dm [hunk 4]
```diff
@@ -358,7 +358,7 @@
 	var/mob/living/carbon/human/H = mymob
 
 	var/mob/screenmob = viewer || H
-
+	world << "DEBUG phil235 viewer: [viewer] ; H: [H] ; screenmob:[screenmob]"
```
Comment (phil235) [inline code/_onclick/hud/human.dm]: fuck

File: code/modules/surgery/bodyparts/helpers.dm [hunk 5]
```diff
@@ -25,7 +25,7 @@
 
 /mob/living/carbon/has_left_hand()
 	for(var/obj/item/bodypart/L in hand_bodyparts)
-		if(!(L.held_index % 2))
+		if(L.held_index % 2)
```
Comment (RemieRichards) [inline code/modules/surgery/bodyparts/helpers.dm]: Why? there's no point to this.

**Comment by AnturK** [comment]
Here we go

**Pull request merged**